client/src/lib.rs

```diff
@@ -63,6 +63,18 @@ const OTHER_CALL_TIMEOUT: Duration = Duration::from_secs(30);
 
 type Result<T, E = tonic::Status> = std::result::Result<T, E>;
 
+/// Options for a namespace-bound connection to the temporal server
+pub struct NamespacedServerGatewayOptions {
+    pub namespace: String,
+    pub opts: ServerGatewayOptions,
+}
```

Comment:
I'm not sure we want this at all.
It might make more sense to just have a single `ServerGateway` or `WorkflowService` that's not bound to a namespace and attach the namespace to worker instances or the higher level rust client for test purposes.

Reply:
I need it to avoid remaking the world for testing. I am totally fine with non-rust langs not using it, or migrating away from it long term.